modules/core/src/main/java/org/apache/ignite/internal/commandline/cache/CheckIndexInlineSizes.java

```diff
@@ -0,0 +1,170 @@
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.commandline.cache;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Predicate;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.apache.ignite.IgniteSystemProperties;
+import org.apache.ignite.internal.client.GridClient;
+import org.apache.ignite.internal.client.GridClientConfiguration;
+import org.apache.ignite.internal.client.GridClientNode;
+import org.apache.ignite.internal.commandline.Command;
+import org.apache.ignite.internal.commandline.cache.check_indexes_inline_size.CheckIndexInlineSizesResult;
+import org.apache.ignite.internal.commandline.cache.check_indexes_inline_size.CheckIndexInlineSizesTask;
+import org.apache.ignite.internal.util.typedef.F;
+import org.apache.ignite.internal.util.typedef.internal.SB;
+import org.apache.ignite.internal.visor.VisorTaskArgument;
+
+import static java.util.stream.Collectors.toSet;
+import static org.apache.ignite.internal.IgniteFeatures.CHECK_INDEX_INLINE_SIZES;
+import static org.apache.ignite.internal.commandline.CommandLogger.INDENT;
+import static org.apache.ignite.internal.commandline.cache.CacheCommands.usageCache;
+
+/**
+ * Command for check secondary indexes inline size on the different nodes.
+ */
+public class CheckIndexInlineSizes implements Command<Void> {
+    /** Warn message format. */
+    public static final String NOT_ALL_NODES_SUPPORT_FEATURE_WARN_MSG_FMT =
+        "Indexes inline size have been checked on limited nodes. Skipped nodes: %s";
+
+    /** Success message. */
+    public static final String INDEXES_INLINE_SIZE_ARE_THE_SAME =
+        "All secondary indexes have the same effective inline size on all cluster nodes.";
+
+    /** Predicate to filter server nodes. */
+    private static final Predicate<GridClientNode> SRV_NODES = node -> !node.isClient() && !node.isDaemon();
+
+    /** {@inheritDoc} */
+    @Override public Object execute(GridClientConfiguration clientCfg, Logger log) throws Exception {
+        try (GridClient client = Command.startClient(clientCfg)) {
+            Set<GridClientNode> serverNodes = client.compute().nodes().stream()
+                .filter(SRV_NODES)
+                .collect(toSet());
+
+            Set<GridClientNode> supportedServerNodes = serverNodes.stream()
+                .filter(n -> n.supports(CHECK_INDEX_INLINE_SIZES))
+                .collect(toSet());
```

Comment:
Is that fine that we omit server nodes that do not support the feature? Than we don't check inline sizes there, isn't that a risk of inconsistency with false-positive check?

Reply:
Though I can see `log.info()` call for that case

Reply:
Already done. Look at CheckIndexInlineSizes#analyzeResults:101 line.